### PR TITLE
Issue #3344237 by kashandarash: Add notification support for config entity

### PR DESCRIPTION
**Pull request by kashandarash**
## Problem
To complete https://getopensocial.atlassian.net/browse/PROD-445 we need to add support for the config entity for the notifications system.

## Solution
Updated activity_creator and activity_logger modules to add support for config entities.
To Make it works config entity must extend NotificationConfigEntityBase class that has needed methods.

## Issue tracker
https://www.drupal.org/project/social/issues/3344237#comment-14939516
https://getopensocial.atlassian.net/browse/PROD-445

## Theme issue tracker
None.

## How to test
- [ ] Make sure the existing notifications works

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
None.

## Release notes
Added notifications support for config entity.

## Change Record
Added abstract class NotificationConfigEntityBase for config entity to extend to have the possibility to use notifications.

## Translations
None.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

File: modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php [hunk 1]
```diff
@@ -97,20 +98,12 @@ public function createMessages(EntityBase $entity, $action) {
...
-      }
-
       // Get the owner or default to anonymous.
       if ($entity instanceof EntityOwnerInterface && $entity->getOwner() !== NULL) {
-        $new_message['uid'] = $entity->getOwner()->id();
+        $new_message['uid'] = (string) $entity->getOwner()->id();
       }
       else {
-        $new_message['uid'] = 0;
+        $new_message['uid'] = '0';
```
Comment (nkoporec) [inline modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php:106]: Why the change to string instead of  the int type ?

**Review by nkoporec — CHANGES_REQUESTED** [review]
Can you update the HTT text ? It should state on how to test this with proper steps that  I can taken in order to check if this is really working.

Also the release notes should be updated, its not really clear to end users why this change was added and what we gain by implementing this.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

File: modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php (hunk 1 above)
Comment (kashandarash) [reply modules/custom/activity_logger/src/Service/ActivityLoggerFactory.php:106]: hello, it is used for "checkIfMessageExist", and $query->condition needs to have string|null as a parameter

**Comment by kashandarash** [comment]
@nkoporec hello, there is no way to check it using only UI, we need to make sure the existing notification system works as before
to check it for config entity you need to create a custom config entity that extends NotificationConfigEntityBase 
Sorry, but are you sure that "How to test" should include instructions on how to develop a config entity?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3321-oo4nrsznbg6oqvvnw3yrc3sibgtrgzmg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/63f89566f8899e08706e3d43

**Review by nkoporec — APPROVED** [review]
I checked the notifications on the Tugboat env. and they are working, so approving this as it is an additional feature and it doesn't impact existing notifications.

**Comment by nkoporec** [comment]
:cherries: picked to 11.8.x
